kili/helpers.py

```diff
@@ -355,3 +359,77 @@ def validate_category_search_query(query):
         expr.parseString(query, parseAll=True)
     except pp.ParseException as error:
         raise ValueError(f"Invalid category search query: {query}") from error
+
+
+def get_file_paths_to_upload(files: Tuple[str, ...], input_type: str, verbose: bool) -> List[str]:
+    """Get a list of paths for the files to upload given a list of files or folder paths.
+
+    Args:
+        files: a list path that can either be file paths, folder paths or unexisting paths
+        input_type: input type of the project to import data to.
+
+    Returns:
+        a list of the paths of the files to upload, compatible with the project type.
+    """
+    file_paths = []
+    for item in files:
+        if os.path.isfile(item):
+            file_paths.append(item)
+        elif os.path.isdir(item):
+            folder_path = os.path.join(item, "")
+            file_paths.extend(
+                [sub_item for sub_item in glob.glob(folder_path + "*") if os.path.isfile(sub_item)]
+            )
+        else:
+            file_paths.extend(
+                [sub_item for sub_item in glob.glob(item) if os.path.isfile(sub_item)]
+            )
+
+    if input_type == "LABEL":
```

Comment:
Quite dirty... We again mix one argument with several types...
It could work if we don't generalize this function to other files even if not beautiful...
The better way would be to give a filtering function as argument to the get_file_paths_to_upload when calling it.
Either do this or write in the docstrings that input_type should be the input_type of a project or the name "LABEL" when importing labels. It will help future dev to understand what you did

Reply:
Still not perfect with the last commit but I hope good enough...